cmd/ocm-backplane/managedJob/createManagedJob.go

```diff
@@ -193,31 +213,94 @@ func waitForCreateJob(client BackplaneApi.ClientInterface, job *BackplaneApi.Job
 	pollErr := wait.PollImmediate(10*time.Second, time.Duration(600)*time.Second, func() (bool, error) {
 		fmt.Printf(".")
 
-		// ========= Get the current job ============
-		jobResp, err := client.GetRun(context.TODO(), options.clusterId, *job.JobId)
-
+		// Get the current job
+		status, err := getJobStatus(client, job)
 		if err != nil {
 			return false, err
 		}
 
-		formatJobResp, err := BackplaneApi.ParseGetRunResponse(jobResp)
+		// Check if the job is in the expected status
+		if status == BackplaneApi.JobStatusStatusSucceeded {
+			statusMessage = "Job Succeeded"
+			return true, nil
+		}
+		if status == BackplaneApi.JobStatusStatusFailed {
+			statusMessage = "Job Failed"
+			return true, nil
+		}
+
+		return false, nil
+	})
+
+	return statusMessage, pollErr
+}
+
+// fetchJobLogs stream the log of the job to the console output when the job status is Running, Succeeded or Failed
+func fetchJobLogs(client BackplaneApi.ClientInterface, job *BackplaneApi.Job) error {
 
+	pollErr := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
I guess you need to increase time out to >1min, otherwise, this func will call 12 times and timeout.
I mean if the job is running for more than 1 min then this function will timeouts.

Reply:
I think it should be waiting for 1 minute till the job status changes to Running/Completed/Failed ?

Reply:
My understanding is `wait.PollImmediate`[1] calls the function each 5 sec and waits 1 minute to timeout. 
[1] https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediate

I think you are right, if the status is not changed within 1 min then it may be related to other issues.

Reply:
yes, it is. It will wait up to 1 minute, but in the whole wait function, it just need to wait for the status to be expected.
The `GetJobLogs` is out of the wait function.